packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -274,6 +274,7 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         }
         this._disposed = true;
 
+        this.removeAllListeners();
```

Comment:
this should happen after we emit dipose, otherwise we will remove any possible handler before we emit